File: src/cujae/inf/citi/om/tools/Tools.java
package cujae.inf.citi.om.tools;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Random;


public class Tools {

	public static double roundDouble(double number, int decimalPlace){
		double numberRound;
		BigDecimal bd = new BigDecimal(number);
		bd = bd.setScale(decimalPlace, BigDecimal.ROUND_UP);
		numberRound = bd.doubleValue();
		return numberRound;
	}

	public static void OrdenateMethod(ArrayList<Double> listCapacities, OrderType typeOrder){
		boolean flag = false;
		
		switch(typeOrder.ordinal())
		{
			case 0: case 1:
			{
				for(int i = 0; i < listCapacities.size() - 1; i++)
				{
					double min = listCapacities.get(i); 
					int pos = i;
	
					for(int j = (i + 1); j < listCapacities.size(); j++)
					{
	
						if(typeOrder.ordinal() == 0)
						{
							if(listCapacities.get(j) <= min)
							{
								min = listCapacities.get(j);
								pos = j;
								
								flag = true;
							}
						}
	
						else
						{
							if(listCapacities.get(j) >= min) 
							{
								min = listCapacities.get(j);
								pos = j;
								
								flag = true;
							}
						}
	
						if(flag)
						{
							listCapacities.set(pos, listCapacities.get(i));
							listCapacities.set(i, min);
							
							flag = false;
						}
					}
				}    
				break;
			}
			case 2:
			{
				Random random = new Random ();
				int index = -1;
	
				ArrayList<Double> listCapacityOrder = new ArrayList<Double>();
	
				int i = 0;
				while(i < listCapacities.size())
				{
					index = random.nextInt(listCapacities.size());
					listCapacityOrder.add(listCapacities.get(index));
					listCapacities.remove(index);
				}
	
				listCapacities.addAll(listCapacityOrder);
				
				break;
			}
		}
	}
}

File: _BHCVRP_Python_Version/cujae_inf_citi_om/tools/Tools.py
import random
from decimal import Decimal, ROUND_UP
from tools.OrderType import OrderType

class Tools:
    @staticmethod
    def round_double(number, decimal_place):
        bd = Decimal(str(number))
        rounded = bd.quantize(Decimal(f'0.{"0"*decimal_place}'), rounding=ROUND_UP)
        return rounded.to_float()

    @staticmethod
    def ordinate_method(list_capacities, type_order: OrderType):
        flag = False

        if type_order == 0 or type_order == 1 or type_order == OrderType.Ascending or type_order == OrderType.Descending:
            for i in range(len(list_capacities) - 1):
                min_val = list_capacities[i]
                pos = i

                for j in range(i + 1, len(list_capacities)):
                    if type_order == 0:
                        if list_capacities[j] <= min_val:
                            min_val = list_capacities[j]
                            pos = j
                            flag = True
                    else:
                        if list_capacities[j] >= min_val:
                            min_val = list_capacities[j]
                            pos = j
                            flag = True

                    if flag:
                        list_capacities[pos], list_capacities[i] = list_capacities[i], min_val
                        flag = False
        elif type_order == 2 or type_order == OrderType.Random:
            list_capacity_order = []
            while list_capacities:
                index = random.randint(0, len(list_capacities) - 1)
                list_capacity_order.append(list_capacities.pop(index))
            list_capacities.extend(list_capacity_order)

